core-rust/state-manager/src/mempool/simple_mempool.rs

```diff
@@ -120,39 +133,139 @@ impl MempoolData {
     }
 }
 
+/// A wrapper around an [`Arc<MempoolData>`] which implements ordering traits for the proposal priority.
+/// The order is such that a normal iteration over a BTreeSet gives the next best proposal transactions,
+/// i.e. if a < b then a is prioritized over b.
+#[derive(Clone, Eq)]
```

Comment:
I'm quite scared about the performance of the `Eq` operator here, particularly if it's used `O(N log N)` times - can we derive a more explicit `Eq` on `MempoolData` or `MempoolTransaction` to use the transaction hash?

Reply:
Thanks for catching this. I wanted to do it but forgot :(